src/components/header/Header.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import SearchBox from "../searchbox/SearchBox";
+import ProfileMenu from "../../pages/homepage/ProfileMenu";
+import { useNavigate } from "react-router-dom"; // Import useNavigate
+import { toast } from "sonner";
+import styled from "styled-components";
+
+export default function Header(props: { onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void }) {
+
+  const navigate = useNavigate(); // For navigating after logout
+
+  // Check for user token or username in local storage
+  const isLoggedIn = localStorage.getItem("token");
+  const username = localStorage.getItem("username");
+
+  const handleLogout = async () => {
+    const token = localStorage.getItem("token");
+
+    // Check if the token exists
+    if (!token) {
```

Comment:
Can't it be just  `if (!token) return toast.error("No token found");` ?

Reply:
OK